### DISCUSSION TITLE
Where can I find a list of deprecations for major versions?

### DISCUSSION THREAD

**Discussion by digiwand** [poteto/ember-changeset#570] [Q&A]
I'm on the way to upgrading `ember-changeset` and `ember-changeset-validations` from `1.x -> 2.x` and `2.x -> 3.x`. I'm having trouble identifying what changes need to be applied to my codebase. I deduced from the 2.x documentation that in order to upgrade from `1.x -> 2.x`, I needed to utilize the new `changeset-get` and `changeset-set` helpers where applicable... I'm not sure if I'm missing anything else, but it seems to be working for the most part. Now that I'm upgrading to `3.x`, my application seems to be broken again and I don't know what I need to change. I hope someone is available to assist me with this despite being in these chaotic times (COVID-19 pandemic) 🤞

**Comment by snewcomer** [comment]
👋  Is your application entirely broken or are some features not working?  It would be good to see a screenshot of the error in the browser dev tools.  You can also use the stop sign to walk through the errors as the application loads to see if it stop on a line involving ember-changeset.

https://github.com/poteto/ember-changeset/blob/master/CHANGELOG.md#change-log

![Screen Shot 2021-01-06 at 8 31 03 AM](https://user-images.githubusercontent.com/7374640/103779608-9dedf200-4ff9-11eb-8344-b988453d3c7b.png)

**Reply by digiwand** [reply]
Hi 👋 It's only part of the feature that uses these libraries that is broken.  I'm not seeing any errors in the console. We have a form on our website. When we click "Submit", it will run validations from `validator` and print out error messages. 

We actually retrieve the error messages like so... 

```
{{#each model.error.action.actions.validation.firstObject as |itemErrors|}}
  {{#unless (eq itemErrors true)}}
  <div class="WorkflowEngineError">
    - {{itemErrors}}
  </div>
  {{/unless}}
{{/each}}
```

When upgrading to `2.x`, the error messages display, but upon updating the inputs, the error messages remain when they should disappear.

When upgrading to `3.x`, the error messages don't work at all...

Thanks for sending over the changelog. I've seen it before, but I was crossing my fingers for an explicit list of deprecations. It's good to know that this is probably the best source of reference.
